app/services/concerns/curator/filestreams/attacher.rb

```diff
@@ -0,0 +1,186 @@
+
+          {
+            io: content_io(io_hash),
+            filename: attributes['file_name'],
+            content_type: attributes['content_type'],
+            metadata: attributes['metadata']
+          }
+        end
+
+        def file_attributes(attachment = {})
+          attachment.slice('created_at', 'file_name', 'content_type', 'byte_size', 'checksum_md5').merge('metadata' => attachment.fetch('metadata', {}))
+        end
+
+        def fedora_content?(io = {})
+          io.dig('fedora_content_location').present?
+        end
+
+        def base64_content?(io = {})
+          io.dig('base64_content').present?
+        end
+
+        def ingest_content?(io = {})
+          io.dig('ingest_filepath').present?
+        end
+
+        private
+
+        # @param attachment [Hash]
+        # @param io [Tempfile]
+        # @returns [ActiveStorage::Blob]
+        def import_file_from_fedora(attachment_attributes, io)
+          blob = ActiveStorage::Blob.create_before_direct_upload!(
+            filename: attachment_attributes['file_name'],
+            byte_size: attachment_attributes['byte_size'],
+            checksum: attachment_attributes['checksum'],
+            content_type: attachment_attributes['content_type'],
+            metadata: attachment_attributes['metadata']
+          )
+          blob.upload_without_unfurling(io)
+          blob
+        end
+
+        def file_set_attachments
+          file_set_class.attachment_reflections.keys
+        end
+      end
+    end
+
+    module AttacherUtils
+      extend ActiveSupport::Concern
+
+      included do
+        include InstanceMethods
+      end
+
+      module InstanceMethods
+        protected
+
+        # format legacy datastream names as ActiveStorage Attachement types
+        def attachment_for_ds(datastream_name)
+          return if datastream_name.blank?
+
+          attachment_type = datastream_name.underscore
+          attachment_type.insert(-4, '_') if datastream_name.match?(/(300|800|marcxml)\z/)
+          attachment_type
+        end
+
+        # use fedora content URL if available, or local filepath, or base64 string
+        def content_io(io_hash)
+          return if io_hash.blank?
+
+          return fedora_io(io_hash['fedora_content_location']) if fedora_content?(io_hash)
+
+          return base64_io(io_hash['base64_content']) if base64_content?(io_hash)
+
+          return file_path_io(io_hash['ingest_filepath']) if ingest_content?(io_hash)
+
+          raise ActiveRecord::RecordNotSaved, "Unknown content for io #{type}"
+        end
+
+        # check ActiveStorage blob attributes against Fedora-exported data, throw error if mismatch
+        # @param blob [ActiveStorage::Blob]
+        # @param byte_size [Integer] size from incoming file
+        # @param checksum_md5 [String] md5 checksum from incoming file
+        def check_file_fixity(blob, byte_size, checksum_md5)
```

Comment:
This method is no longer used anywhere, is that intentional? It used to be one of the last steps in `FileFactoryService#call`.